src/cli/shell.rs

```diff
@@ -215,6 +215,8 @@ pub async fn execute(args: Args) -> miette::Result<()> {
 
     // Get the environment variables we need to set activate the environment in the shell.
     let env = get_activation_env(&environment, args.lock_file_usage.into()).await?;
+    let mut env = env.clone();
+    env.insert(String::from("PIXI_IN_SHELL"), String::from("1"));
```

Comment:
I would move this into the `src/activation.rs` `Environment::get_metadata()` function so it is generally available in all activated environment.